### ISSUE TITLE
Description Computation for buttons not consistent

### ISSUE THREAD

**Issue by JAWS-test** [w3c/html-aam#278]
I wonder why the Description Computation is not consistent for button and I can't find any reason for the differences in [ACC Name](https://w3c.github.io/accname/):
* [Button with `input`](https://w3c.github.io/html-aam/#input-type-button-input-type-submit-and-input-type-reset-accessible-description-computation): visual text content can be description (`value`)
* [Button with `summary`](https://w3c.github.io/html-aam/#summary-element-accessible-description-computation): Text content can be description
* [Button with `button`](https://w3c.github.io/html-aam/#button-element-accessible-description-computation): Text content cannot be description

**Comment by scottaohara** [comment]
With Chrome, if you review the accessibility tree of the elements in [this test case](https://scottaohara.github.io/testing/buttons/desc.html), the `value` of an `input type=button` with an `aria-label` is exposed as a description.  As is the visible text of a `summary` with an `aria-label`.

In Webkit the visible text of a `summary` is exposed to the a11y tree (noted as `title` content in Apple's Accessibility Inspector) and can be read by VO if one were to navigate into the `summary` element (`summary` is not exposed as a `button` in Webkit).

Firefox doesn't expose the `value` of an `input type=button` or subtree of a `summary` as description content.

In summary: 
* Chrome and Chromium Edge with JAWS or NVDA will announce the descriptions.
* macOS Safari + VO will allow access to the visible text/content of the `summary`, but doesn't use the `value` of a `input type=button` as description content.
* Firefox doesn't follow the steps of these description algorithms.

**Comment by JAWS-test** [comment]
Regardless of the test results, I wonder if this should not be unified and if Accname should be updated

**Comment by scottaohara** [comment]
Don't see a good way for unification since each of these elements have variations in how they're used.  e.g. `value` is not used for `summary`, and doesn't expose name/desc information for the `button` element.

**Comment by JAWS-test** [comment]
There are two possibilities:
* For all buttons the visible label can become a description
* For no button the visible label is used for description

**Comment by scottaohara** [comment]
the first possible simplification would result in
```html
<button aria-label=close>
  X
</button>
```
having a description of "X".  I highly doubt we'd want to do this.


Which leaves the second option, which does bring these buttons inline with how other elements work.  But I'm curious of any reasoning as to why these were outlined this way to begin with before anything is done here.

**Comment by JAWS-test** [comment]
I would also prefer the second option

**Comment by JAWS-test** [comment]
When I label an `<input type=button>` with aria-label, the visible label in the value automatically becomes the description. However, this is redundant for most buttons, since [WCAG 2.5.3](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html) requires that the visible label be included in the Accessible Name. Therefore, I suggest that value should not be used to determine the description.

Example: `<input type=button value=Delete aria-label="Delete product A">` causes "Delete" to be output twice by the screen reader.
